src/unitxt/inference.py

```diff
@@ -164,86 +185,645 @@ def _is_loaded(self):
         pass
 
 
-class HFPipelineBasedInferenceEngine(
-    InferenceEngine, PackageRequirementsMixin, LazyLoadMixin
+class HFGenerationParamsMixin(Artifact):
+    max_new_tokens: int
+    do_sample: bool = False
+    temperature: Optional[float] = None
+    top_p: Optional[float] = None
+    top_k: Optional[int] = None
+    num_beams: Optional[int] = None
+    repetition_penalty: Optional[float] = None
+    pad_token_id: Optional[int] = None
+    eos_token_id: Optional[int] = None
+
+
+class HFInferenceEngine(
+    InferenceEngine,
+    LogProbInferenceEngine,
+    PackageRequirementsMixin,
+    LazyLoadMixin,
+    HFGenerationParamsMixin,
 ):
     model_name: str
-    max_new_tokens: int
+    label: str
+
+    n_top_tokens: int = 5
+
+    device: Any = None
+    device_map: Any = None
+
+    use_fast_tokenizer: bool = True
     use_fp16: bool = True
```

Comment:
I don't see you use these and support this

Reply:
Right, this should be in pipelines only